addons/stock/models/stock_move_line.py

```diff
@@ -241,36 +241,27 @@ def write(self, vals):
         moves_to_recompute_state = self.env['stock.move']
         Quant = self.env['stock.quant']
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-        # We forbid to change the reserved quantity in the interace, but it is needed in the
-        # case of stock.move's split.
-        # TODO Move me in the update
-        if 'product_uom_qty' in vals:
-            for ml in self.filtered(lambda m: m.state in ('partially_available', 'assigned') and m.product_id.type == 'product'):
-                if not ml._should_bypass_reservation(ml.location_id):
-                    qty_to_decrease = ml.product_qty - ml.product_uom_id._compute_quantity(vals['product_uom_qty'], ml.product_id.uom_id, rounding_method='HALF-UP')
-                    try:
-                        Quant._update_reserved_quantity(ml.product_id, ml.location_id, -qty_to_decrease, lot_id=ml.lot_id, package_id=ml.package_id, owner_id=ml.owner_id, strict=True)
-                    except UserError:
-                        if ml.lot_id:
-                            Quant._update_reserved_quantity(ml.product_id, ml.location_id, -qty_to_decrease, lot_id=False, package_id=ml.package_id, owner_id=ml.owner_id, strict=True)
-                        else:
-                            raise
-
         triggers = [
             ('location_id', 'stock.location'),
             ('location_dest_id', 'stock.location'),
             ('lot_id', 'stock.production.lot'),
             ('package_id', 'stock.quant.package'),
-            ('result_package_id', 'stock.quant.package'),
             ('owner_id', 'res.partner')
         ]
         updates = {}
         for key, model in triggers:
             if key in vals:
                 updates[key] = self.env[model].browse(vals[key])
 
-        if updates:
+        if updates or 'product_uom_qty' in vals:
             for ml in self.filtered(lambda ml: ml.state in ['partially_available', 'assigned'] and ml.product_id.type == 'product'):
+
+                if 'product_uom_qty' in vals:
+                    new_product_uom_qty = ml.product_uom_id._compute_quantity(
+                        vals['product_uom_qty'], ml.product_id.uom_id, rounding_method='HALF-UP')
+                else:
+                    new_product_uom_qty = ml.product_qty
+
                 if not ml._should_bypass_reservation(ml.location_id):
```

Comment:
Is this intended to un-reserve before reserving? I think that would be helpful to have some comments in the code about the purpose of each `_update_reserved_quantity` calls.

Reply:
So if you change, let's say, a lot, we unreserve the original lot and try to reserve the new one.
The original issue is that if you change a lot and the actual reserved quantity in the same call, the quantity is unreserved two times. I think it makes sense to handle a change of reserved quantity the same way we handle a change of lot/owner/location/package. It fixes the original issue "by construction" and the structure of the `write` method is simplified, with only two calls to `_update_reserved_quantity` (each one having a shitty fallback on quants on tracked products without an assigned lot).

Reply:
Yep my comment was more about having a clarification comment in the code since the purpose of each call was not obvious unless one reads the third argument of `_update_reserved_quantity`.